src/Microsoft.AspNet.Hosting/WebApplicationBuilder.cs

```diff
@@ -202,12 +141,12 @@ public WebHostBuilder UseStartup(Type startupType)
             return this;
         }
 
-        public WebHostBuilder UseStartup<TStartup>() where TStartup : class
+        public WebApplicationBuilder UseStartup<TStartup>() where TStartup : class
         {
             return UseStartup(typeof(TStartup));
         }
 
-        public WebHostBuilder UseStartup(Action<IApplicationBuilder> configureApp)
+        public WebApplicationBuilder UseStartup(Action<IApplicationBuilder> configureApp)
```

Comment:
Maybe split the names where UseStartup is for Type/TStartup, and Configure for the overloads that take an configureAction.  Underneath the cover we would still wire it via a StartupMethods, but there's no real reason that name needs to be in the Builder API does it?  Maybe we just kill the UseStartup overload that takes both configureApp and conifgureServices, since they can do UseServices/Configure as replacements